### PR TITLE
[Automation] Docker Image Validation: size regression, visualization, token-based access

### PR DESCRIPTION
**Pull request by AndreyKoltsov1997**
The PR introduces:
* The ability to validate the size of TeamCity Docker Images: automation framework, DSL for build configuration, a related updates within DSL Generation;
* Visualization of image size trend across latest TeamCity images;

It accumulates all of the recent changes that have been done within the scope of this task. Please, note that it does not include the generated DSLs for other build configurations created via C# generator.

### CODE REVIEW

File: .teamcity/generated/ImageValidation.kts [hunk 1]
```diff
@@ -6,116 +6,90 @@ package generated
...
-//                reuseBuilds = ReuseBuilds.ANY
-//                onDependencyFailure = FailureAction.IGNORE
-//            }
-//        }
-//	}
-    })
+object image_validation: BuildType({
+
+	name = "Validation of Size Regression - Staging Docker Images (Windows / Linux)"
+	buildNumberPattern="test-%build.counter%"
```
Comment (burnasheva) [inline .teamcity/generated/ImageValidation.kts:21]: Let's remove test- prefix? I'd suggest using the same pattern as was used for the push configurations (buildNumber-counter)

File: .teamcity/generated/ImageValidation.kts [hunk 2]
```diff
@@ -6,116 +6,90 @@ package generated
...
+		// Execute the build once the images are available within %deployRepository%
+		finishBuildTrigger {
+			buildType = "${PublishLocal.publish_local.id}"
+		}
+	}
+
+	params {
+		// Inherited parameter, not used within build configuration, thus removed in order to ...
+		// ... prevent unnecessary dependency.
+		param("dockerImage.teamcity.buildNumber", "-")
```
Comment (burnasheva) [inline .teamcity/generated/ImageValidation.kts:37]: Let's use it in the build number :)

File: .teamcity/generated/ImageValidation.kts [hunk 3]
```diff
@@ -6,116 +6,90 @@ package generated
...
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-linux",
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-linux-sudo",
+		"%docker.deployRepository%teamcity-minimal-agent%docker.buildImagePostfix%:2022.10-linux"
+	)
+
+	steps {
+		images.forEach { imageFqdn ->
+			// Generate validation for each image fully-qualified domain name (FQDN)
+			gradle {
+				name = "Image Verification Gradle - $imageFqdn"
```
Comment (burnasheva) [inline .teamcity/generated/ImageValidation.kts:57]: This parameter isn't resolved, and a step name looks like 
`Image Verification Gradle - %docker.deployRepository%teamcity-server%docker.buildImagePostfix%:2022.10-nanoserver-2004`

File: .teamcity/generated/ImageValidation.kts [hunk 4]
```diff
@@ -6,116 +6,90 @@ package generated
...
+		// ... Docker agent will be used, platform-compatibility must be addressed, ...
+		// ... especially in case of Windows images.
+		contains("teamcity.agent.jvm.os.name", "Linux")
+	}
+
+	features {
+		dockerSupport {
+			cleanupPushedImages = true
+			loginToRegistry = on {
+				dockerRegistryId = "PROJECT_EXT_774,PROJECT_EXT_315"
```
Comment (burnasheva) [inline .teamcity/generated/ImageValidation.kts:91]: Do we need to log in to both registries here? I thought that only one will be enough

**Review by burnasheva — CHANGES_REQUESTED** [review]

File: .teamcity/generated/ImageValidation.kts (hunk 3 above)
Comment (AndreyKoltsov1997) [reply .teamcity/generated/ImageValidation.kts]: @burnasheva it seems that static fields inside of Build configuration, such as the name of the step, aren't resolved during compile time. I think it is expected, as it's converted into XML within TeamCity Server.
I've addressed the change in https://github.com/JetBrains/teamcity-docker-images/pull/46

File: .teamcity/generated/ImageValidation.kts (hunk 2 above)
Comment (AndreyKoltsov1997) [reply .teamcity/generated/ImageValidation.kts]: Good point, added it within https://github.com/JetBrains/teamcity-docker-images/pull/46

File: .teamcity/generated/ImageValidation.kts (hunk 1 above)
Comment (AndreyKoltsov1997) [reply .teamcity/generated/ImageValidation.kts]: @burnasheva yes, good point, I've changed build name pattern in https://github.com/JetBrains/teamcity-docker-images/pull/46

File: .teamcity/generated/ImageValidation.kts (hunk 4 above)
Comment (AndreyKoltsov1997) [reply .teamcity/generated/ImageValidation.kts]: @burnasheva currently, `PROJECT_EXT_315` is not being used, I've removed it in https://github.com/JetBrains/teamcity-docker-images/pull/46

**Pull request merged**